src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationServicePermissionTests.cs

```diff
@@ -39,6 +39,134 @@ public class PublicationServicePermissionTests
             Title = "Test publication",
         };
 
+        [Fact]
+        public async Task GetMyPublicationsByTopic_NoAccessOfSystem()
+        {
+            var userService = AlwaysTrueUserService();
+            var publicationRepository = new Mock<IPublicationRepository>(Strict);
+
+            var topicId = Guid.NewGuid();
+
+            var publicationService = BuildPublicationService(
+                context: Mock.Of<ContentDbContext>(Strict),
+                userService: userService.Object,
+                publicationRepository: publicationRepository.Object);
+
+            userService.Setup(s => s.MatchesPolicy(SecurityPolicies.CanAccessSystem)).ReturnsAsync(false);
+
+            var list = new List<Publication> {
+                new()
+                {
+                    Id = Guid.NewGuid()
+                }
+            };
+
+            publicationRepository.Setup(s => s.GetPublicationsForTopic(topicId)).ReturnsAsync(list);
+
+            var result = await publicationService.GetPublicationsByTopic(permissions: false, topicId);
+            result.AssertForbidden();
+
+            userService.Verify(s => s.GetUserId());
+            userService.Verify(s => s.MatchesPolicy(SecurityPolicies.CanAccessSystem));
+            userService.VerifyNoOtherCalls();
+
+            publicationRepository.VerifyNoOtherCalls();
+        }
```

Comment:
I'm wondering why some of these tests are so verbose.

For example, couldn't this just be as follows?

```
        [Fact]
        public async Task GetMyPublicationsByTopic_NoAccessOfSystem()
        {
            await PermissionTestUtils.PolicyCheckBuilder<SecurityPolicies>()
                .ExpectCheckToFail(SecurityPolicies.CanAccessSystem)
                .AssertForbidden(async userService =>
                {
                    var service = BuildPublicationService(
                        context: Mock.Of<ContentDbContext>(Strict),
                        userService: userService.Object);
                    return await service.GetPublicationsByTopic(permissions: false, Guid.NewGuid());
                });
        }
```

Reply:
I was copying existing tests and adapting them to my purposes. I'll try to change all of them to use `PolicyCheckBuilder`.

Reply:
So I've removed most of the tests related to `ListPublicationsByTopic` and `GetMyPublicationsAndReleasesByTopic`. The thought process is that the `*PermissionTests.cs` files are for checking that permission checks fail in certain circumstances, causing an endpoint to return a `Left`. Since other permissions checks done in these methods don't cause the endpoint to return a `Left`, that the correct code path is followed is tested by the `PublicationServiceTests.cs`, and don't need an associated permissions test.

Others happy with this?